src/resource/solidDataset.ts

```diff
@@ -844,7 +855,11 @@ export function getContainedResourceUrlAll(
   // See https://www.w3.org/TR/2015/REC-ldp-20150226/#h-ldpc-http_post:
   // > a containment triple MUST be added to the state of the LDPC whose subject is the LDPC URI,
   // > whose predicate is ldp:contains and whose object is the URI for the newly created document
-  return container !== null ? getIriAll(container, ldp.contains) : [];
+  return container !== null
+    ? getIriAll(container, ldp.contains).filter((child) =>
+        isChildResource(child, getSourceUrl(solidDataset))
+      )
+    : [];
```

Comment:
To me it seems more sensible to do the following (factor out the `getSrcUrl` for performance; and error rather than filtering so that users receive errors to alert them to a malicious Pod rather than results just going missing).

```suggestion
  if (container === null) return [];
  const containment = getIriAll(container, ldp.contains);
  // Make a const outside the loop so we only compute this once
  const srcUrl = getSourceUrl(solidDataset);
  
  for (const child of containment) {
    if (!isChildResource(child, srcUrl))
      // Error rather than just hiding the resource
      throw new Error(`Child resource [${child}] not in container [${srcUrl}] according to slash semantics`);
  }
```

Reply:
Agreed, it sounds like throwing here would be a good idea, in particular since there is a direct action the user can take to mitigate the issue: 2e7bebf27d68f188b43ef873cc42bcb874c9c37a